multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -2812,6 +2817,79 @@ def load(
 
         return predictor
 
+    def dump_timm_image(
+        self,
+        path: str,
+    ):
+        """
+        Save TIMM image model weights and config to local directory.
+        Model weights are saved in file `pytorch_model.bin`;
+        Configs are saved in file `config.json`
+
+        Parameters
+        ----------
+        path : str
+            Path to directory where models and configs should be saved.
+        """
+        model = None
+        if isinstance(self._model, MultimodalFusionMLP) and isinstance(
+            self._model.model, torch.nn.modules.container.ModuleList
+        ):
+            for per_model in self._model.model:
+                if isinstance(per_model, TimmAutoModelForImagePrediction):
+                    model = per_model
+                    break
+
+        model = self._model if model is None else model
+        if isinstance(model, TimmAutoModelForImagePrediction) and model.model is not None:
+            os.makedirs(path)
+            weights_path = f"{path}/pytorch_model.bin"
+            torch.save(model.model.state_dict(), weights_path)
+            logger.info(f"Model weights saved to {weights_path}.")
+            config_path = f"{path}/config.json"
+            save_timm_config(model, config_path)
+        else:
+            raise NotImplementedError("Only TIMM image models are supported for dump.")
+
+    def dump_hf_text(
+        self,
+        path: str,
+    ):
+        """
+        Save HuggingFace Text model weights, config and tokenizers to local directory.
+        Model weights are saved in file `pytorch_model.bin`;
+        Configs are saved in file `config.json`
+
+        Parameters
+        ----------
+        path : str
+            Path to directory where models and configs should be saved.
+        """
+        model = None
+        if isinstance(self._model, MultimodalFusionMLP) and isinstance(
+            self._model.model, torch.nn.modules.container.ModuleList
+        ):
+            for per_model in self._model.model:
+                if isinstance(per_model, HFAutoModelForTextPrediction):
+                    model = per_model
+                    break
+
+        model = self._model if model is None else model
+        if isinstance(model, HFAutoModelForTextPrediction) and model.model is not None:
+            os.makedirs(path)
```

Comment:
Try to use “os.makedirs(path, exist_ok=True)”.

Reply:
I originally had this flag set, but after discussing with Zhiqiang offline, he suggested that we might not want saving to an existing (or non-empty) directory to avoid potential override of the model.